diff -u -r -N a/blocky/config/custom_dns.go b/blocky/config/custom_dns.go
--- a/blocky/config/custom_dns.go	2026-06-21 15:34:37.902388628 -0700
+++ b/blocky/config/custom_dns.go	2026-06-21 16:12:28.894576665 -0700
@@ -17,15 +17,23 @@
 	CustomTTL Duration `default:"1h" yaml:"customTTL"`
 	// Simple domain-to-IP mappings; multiple IPs per domain separated by commas.
 	Mapping CustomDNSMapping `yaml:"mapping"`
+	// Reverse mapping override from IP-to-domain.
+	ReverseMappingOverride CustomDNSReverseMappingOverride `yaml:"reverseMappingOverride"`
 	// DNS zone file content for more complex record definitions (A, AAAA, CNAME, TXT, SRV).
 	Zone ZoneFileDNS `default:"" yaml:"zone"`
 	// If true, queries for types not defined for a domain return empty; if false, they are forwarded upstream.
 	FilterUnmappedTypes bool `default:"true" yaml:"filterUnmappedTypes"`
 }
 
+type CustomDNSReverseIP struct {
+	IP dns.RR
+}
+
 type (
 	CustomDNSMapping map[string]CustomDNSEntries
 	CustomDNSEntries []dns.RR
+	CustomDNSReverseMappingOverride map[CustomDNSReverseIP]CustomDNSReverseEntries
+	CustomDNSReverseEntries []string
 
 	ZoneFileDNS struct {
 		RRs        CustomDNSMapping
@@ -70,6 +78,39 @@
 	return nil
 }
 
+func (c *CustomDNSReverseEntries) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var input string
+	if err := unmarshal(&input); err != nil {
+		return err
+	}
+
+	parts := strings.Split(input, ",")
+	result := make(CustomDNSReverseEntries, len(parts))
+
+	for i, part := range parts {
+		result[i] = part
+	}
+
+	*c = result
+
+	return nil
+}
+
+func (c *CustomDNSReverseIP) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var input string
+	if err := unmarshal(&input); err != nil {
+		return err
+	}
+
+	result, err := configToRR(input)
+	if err != nil {
+		return err
+	}
+	*c = CustomDNSReverseIP{ IP: result }
+
+	return nil
+}
+ 
 func (c *CustomDNSEntries) UnmarshalYAML(unmarshal func(any) error) error {
 	var input string
 	if err := unmarshal(&input); err != nil {
diff -u -r -N a/blocky/resolver/custom_dns_resolver.go b/blocky/resolver/custom_dns_resolver.go
--- a/blocky/resolver/custom_dns_resolver.go	2026-06-21 15:34:37.910388855 -0700
+++ b/blocky/resolver/custom_dns_resolver.go	2026-06-21 16:12:52.451155070 -0700
@@ -59,10 +59,25 @@
 	}
 
 	reverse := make(map[string][]string, len(dnsRecords))
+	reverseOverride := make(map[string]bool, len(cfg.ReverseMappingOverride))
+
+	// Add the specified reverse mapping override entries first as-is.
+	for entry, urls := range cfg.ReverseMappingOverride {
+		if ip := extractIPFromRecord(entry.IP); ip != nil {
+			reverseOverride[ip.String()] = true
+			r, _ := dns.ReverseAddr(ip.String())
+			for _, url := range urls {
+				reverse[r] = append(reverse[r], url)
+			}
+		}
+	}
 
 	for url, entries := range dnsRecords {
 		for _, entry := range entries {
-			if ip := extractIPFromRecord(entry); ip != nil {
+			// Only if a given entry doesn't exist in the reverse mapping
+			// override, add the reverse mapping records using the
+			// forward records.
+			if ip := extractIPFromRecord(entry); ip != nil && !reverseOverride[ip.String()] {
 				r, _ := dns.ReverseAddr(ip.String())
 				reverse[r] = append(reverse[r], url)
 			}
